src/GrasshopperTeklaDrawingLink/Components/Text/Arrow component.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Grasshopper.Kernel;
+
+using GTDrawingLink.Properties;
+using GTDrawingLink.Tools;
+using GTDrawingLink.Types;
+
+using System.Collections.Generic;
+using System.Drawing;
+
+using Tekla.Structures.Drawing;
+
+namespace GTDrawingLink.Components.Text {
+    public class ArrowAttributeComponent : TeklaComponentBase {
+
+
+        public ArrowAttributeComponent()
+            : base(ComponentInfos.ArrowAttributesComponent) {
+        }
+
+        public override GH_Exposure Exposure => GH_Exposure.primary;
+        protected override Bitmap Icon => Resources.MoveView;
+
+
+        protected override void RegisterInputParams(GH_InputParamManager pManager) {
+            SetParametersAsOptional(pManager, new List<int> {
+            pManager.AddParameter(new EnumParam<ArrowheadTypes>(ParamInfos.ArrowType, GH_ParamAccess.item))
+            });
+            AddNumberParameter(pManager, ParamInfos.Width, GH_ParamAccess.item, true);
+            AddNumberParameter(pManager, ParamInfos.Height, GH_ParamAccess.item, true);
+        }
+
+        protected override void RegisterOutputParams(GH_OutputParamManager pManager) {
+            pManager.AddParameter(new ArrowAttributesParam(ParamInfos.ArrowAttribute, GH_ParamAccess.item));
+        }
+
+        protected override void SolveInstance(IGH_DataAccess DA) {
```

Comment:
I know that the standard way of writing GH components is not the best sample of single responsibility principle, but this method is quite hard to read due to mixing of getting the input values and constructing the output at the same time.

You are loading each of inputs, checks if it is default, setting some special values, which is needed, I get it, but in the same time you are populating properties of ArrowheadAttributes. If we would count 'if' branches then there is a plenty way how this ArrowheadAttributes would be created. A lot of places to hide some bug.

Please, separate this into two separate sections: getting inputs from IGH_DataAccess (here will be default values) and separate creation of ArrowheadAttributes with setting its properties.